workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/RequiredFieldInjector.cs

```diff
@@ -68,6 +68,26 @@ private IInjectable Inject(MonoBehaviour behaviour, InjectableId injectableId, E
             return injectable;
         }
 
+        public void DisposeAllRequiredFields(MonoBehaviour behaviour)
+        {
+            var behaviourType = behaviour.GetType();
+            EnsureLoaded(behaviourType);
+            foreach (var fieldsToComponents in fieldInfoCache[behaviourType])
+            {
+                var fields = fieldsToComponents.Value;
+                foreach (var field in fields)
+                {
+                    var oldField = field.GetValue(behaviour);
+                    if (!(oldField is RequirableBase))
```

Comment:
Case it once and save the result so you don't have to cast again below.

Reply:
Fixed.